### ISSUE TITLE
Navigator.pushNamedAndRemoveUntil ，The phone may have a black screen

### ISSUE THREAD

**Issue by AlanGrey** [flutter/flutter#29283]
I am using MOTO Phone to test , when I use Navigator.pushNamedAndRemoveUntil api to remove all the routes , and then new screen use Navigator.pop  or backspace key , The phone went black screen.

flow:
1. screen_1 => Navigator.of(context).pushNamed("/screen_1");
2. screen_2 => Navigator.of(context).pushNamed("/screen_2");
3. screen_3 => Navigator.of(context).pushNamedAndRemoveUntil("/screen_4",(Route<dynamic> route) => false);
4. screen_4 => Navigator.of(context).pop();

```
class RouteDemoApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'RouteDemo',
      theme: ThemeData(
        primaryColor: Colors.redAccent,
      ),
      home: RouteUserDemo(),
      routes: <String, WidgetBuilder>{
        '/screen1': (BuildContext context) => new Screen1(),
        '/screen2': (BuildContext context) => new Screen2(),
        '/screen3': (BuildContext context) => new Screen3(),
        '/screen4': (BuildContext context) => new Screen4(),
      },
    );
  }
}
```

**Comment by Kartik1607** [comment]
Just to understand correctly, instead of black screen the application should close. Correct?

**Comment by AlanGrey** [comment]
> Just to understand correctly, instead of black screen the application should close. Correct?

Yes, the normal situation is that the app is closed, but there is a black screen on the moto phone, you need to click again to exit the app.

**Comment by AlanGrey** [comment]
> Just to understand correctly, instead of black screen the application should close. Correct?

I tried to set the MaintainState property of MaterialPageRoute to False and found that it can solve this problem; but this is the way of dynamic route, I hope that the static way can also solve this problem.
